tests/L0/run_transformer/test_pipeline_parallel_fwd_bwd.py

```diff
@@ -108,7 +109,9 @@ def _forward_backward_test_impl(
                 PipelineParallelForwardBackwardTest.HIDDEN_SIZE,
             )
 
-            batch =(((self.rank + 1) * torch.ones(global_batch_shape)).cuda(), )
+            batch = None
+            if parallel_state.is_pipeline_first_stage:
+                batch = (torch.ones(global_batch_shape, dtype=torch.float).cuda(), )
```

Comment:
This is all right if we are 100% certain that this will not cause any trouble if `recv` fails.

Reply:
We would want that to raise an error in any case, it would not be expected behavior. In the current implementation, `batch` should not be used by the downstream stages anyway so this check just enforces this property.